ref/api-reference.md

```diff
@@ -1,7 +1,174 @@
-The documentation in this section provides a reference for the following APIs provided by Anbox Cloud:
+All communication between AMS and its clients happen using a RESTful API over HTTP. This API is encapsulated over either TLS (for remote operations) or a Unix socket (for local operations).
+
+## APIs and their structure
+Anbox Cloud provides the following APIs:
+
 * AMS HTTP API
 * Anbox Cloud HTTP API
 * Stream Gateway API
 * Anbox Platform SDK API
 
-Also see [SDKs](https://discourse.ubuntu.com/t/anbox-cloud-sdks/17844) for a more detailed description of the SDKs that Anbox Cloud provides.
+All these APIs except for the [Anbox HTTP API](https://discourse.ubuntu.com/t/anbox-http-api/17819) have an auto-generated Swagger specification describing its API endpoints.
+
+## Authentication
+
+Not all REST API endpoints require authentication, for example, the following API calls are allowed for everyone:
+
+* `GET /`
+* `GET /1.0`
+* `GET /1.0/version`
+
+Some endpoints require an additional authentication token to ensure that the requester is authorised to access the resource, for example:
+
+* `GET /1.0/artifacts`
+* `PATCH /1.0/instances/<name>`
+
+## API versioning
+The details of a version of the API can be retrieved using `GET /<version>`. For example, `GET /1.0`.
+
+If an API version is bumped to a major version, it indicates that backward compatibility is affected.
+
+Feature additions done without breaking backward compatibility only result in addition to `api_extensions` which can be used by the client to check if a given feature is supported by the server.
+
+## Return values
+There are three standard return types:
+
+ * Standard return value
+ * Background operation
+ * Error
+
+### Standard return value
+For a standard synchronous operation, the following dict is returned:
+
+```json
+{
+    "type": "sync",
+    "status": "Success",
+    "status_code": 200,
+    "metadata": {}               # Extra resource/action specific metadata
+}
+```
+
+HTTP code must be 200.
```

Comment:
Why 'must'? It's not something the user has any control over, and it's not a specification to follow either.
How about 'is'?

Also, I think the correct term is 'HTTP response status code', see https://developer.mozilla.org/en-US/docs/Web/HTTP/Status and https://httpwg.org/specs/rfc9110.html#overview.of.status.codes

```suggestion
HTTP response status code is 200.
```

Wdyt?